docs/application/web/guides/applications/addon.md

```diff
@@ -0,0 +1,200 @@
+  To generate a package for an addon, you can use Tizen Studio. There is a separate tag for addon in config.xml.
+
+- [Management](#management)
+
+  To manage installed addons, you can use a separate Web application named Web App Addon Setting.
+
+<a name="configure"></a>
+## Configuration
+
+### Manifest file (mandatory)
+
+The manifest file includes the metadata for an addon:
+
+  - **name**: Name of the addon.
+  - **version**: Version of the addon.
+  -	**main**: The file name of the main JavaScript.
+  -	**settings**: Not available.
+  -	**description**: Description of the addon.
+
+Following is an example of a manifest file. The filename must be `manifest.json`:
+ 
+    ```
+    {
+        “name”: “addonSampleAd”,
+        "version”: “0.0.1”,
+        “main”: “addonSampleAd”,
+        “settings”: “”,
+        “description”: “Sample to show how to show up advertisement at prelaunch event”
+    }
+    ```
+
+### Main JavaScript File
+
+You must have a main JavaScript file.
+
+The main JavaScript file must have the same name with `main` in the manifest file. The main JavaScript file includes:
+
+  - `activate()`: To register a callback to an addon event.
+  - `deactivate()`: To unregister the callback to the addon event.
+  - `Callback Functions`: You can write your own code to define the behavior that happens when an addon event occurs.
+
+Following is an example of a main JavaScript file:
+
+    ```
+    class AddonSampleAd {
+        constructor(em) {
+          console.log('AddonSampleAd constructor');
+          this.evt = em;
+        }
+        activate() {
+            console.log('AddonSampleAd.activate');
+            this.evt.on('lcPrelaunch', this.prelaunch);
+        }
+        deactivate() {
+            console.log('AddonSampleAd.dectivate');
+            this.evt.off('lcPrelaunch', this.prelaunch);
+        }
+        prelaunch(winId, orgUrl) {
+            console.log('AddonSampleAd.prelaunch');
+        }
+    }
+    module.exports = AddonSampleAd;
+
+    ```
+
+<a name="event"></a>
+## Event
+
+| Event                                      | Description                              |
+| ------------------------------------------ | ---------------------------------------- |
+| lcPrelaunch | Emitted once for Web app launching just before the main page is shown. |
+| lcResume | Emitted when the app is resumed. |
+| lcSuspend | Emitted when the app is suspended. |
+| lcQuit | Emitted before the app is ended. |
+| contentDidFinishLoad | Emitted when the page navigation is done. | 
+
+<a name="mechanism"></a>
+## Mechanism
+
+Addons are not standalone software programs, and only run when a Web application runs. They work on event-driven invocation. Each addon registers its callbacks to events that are pre-defined in Web Runtime JS (WRTjs). Registered callbacks are invoked whenever the registered events occur during the Web application’s life cycle.  
+
+**Figure: Addon Mechanism**
+
+![Addon mechanism](./media/addon_mechanism.png)
```

Comment:
In the image, how about "Start events" instead of "Emit events"?

Reply:
With the same reason I mentioned for using 'Emitted' in the table above, I prefer to using 'Emit'. Depending on the agreement of it, I will change the word or not.

Reply:
In my opinion too "Emit" is better to use here.

Reply:
Thanks for the answer and the opinion. I also agree on using `Emit`.